rules/Php74/Guard/PropertyTypeChangeGuard.php

```diff
@@ -76,15 +76,10 @@ private function isSafeProtectedProperty(ClassReflection $classReflection, Prope
             return false;
         }
 
-        $class = $this->classLikeAstResolver->resolveClassFromClassReflection($classReflection);
-        if (! $class instanceof Class_) {
+        if (! $classReflection->isFinalByKeyword()) {
```

Comment:
`isFinalByKeyword()` here is needed to ensure read final from class, not docblock, same with `$class->isFinal()` usage which read from real `final` class.